ethereum/core/src/main/java/org/hyperledger/besu/ethereum/trie/diffbased/common/worldview/DiffBasedWorldState.java

```diff
@@ -165,8 +166,23 @@ public void persist(final BlockHeader blockHeader) {
 
     boolean success = false;
 
-    final DiffBasedWorldStateKeyValueStorage.Updater stateUpdater =
-        worldStateKeyValueStorage.updater();
+    // Bonsai archive uses context set in the world state storage for the specified block.
+    // Update the block context before putting entries to storage via calculateRootHash()
+    // TODO - rename calculateRootHash() to be clearer that it updates state, it doesn't just
+    // calculate a hash
+    if (worldStateKeyValueStorage.getFlatDbStrategy() instanceof BonsaiArchiveFlatDbStrategy
```

Comment:
I prefer to put it all the time, it doesn't really have a huge impact and avoids an additional condition